mailparse.c

```diff
@@ -172,57 +178,52 @@ PHP_MSHUTDOWN_FUNCTION(mailparse)
 
 static inline php_mimepart *mimemsg_get_object(zval *object TSRMLS_DC)
 {
-	zval **zpart;
+	zval *zpart;
 	php_mimepart *part;
-	int type;
-	
 
 	if (Z_TYPE_P(object) != IS_OBJECT) {
 		return NULL;
 	}
 
-	if (zend_hash_index_find(Z_OBJPROP_P(object), 0, (void**)&zpart) == FAILURE) {
+	if ((zpart = zend_hash_index_find(Z_OBJPROP_P(object), 0)) == NULL) {
 		return NULL;
 	}
 
-	part = zend_list_find(Z_LVAL_PP(zpart), &type);
-
-	if (type != le_mime_part)
+	if ((mailparse_fetch_mimepart_resource(part, zpart)) == NULL) {
 		return NULL;
+	}
 
 	return part;
 }
 
 static int mailparse_mimemessage_populate(php_mimepart *part, zval *object TSRMLS_DC)
 {
-	zval *tmp;
+	zval tmp;
 
-	MAKE_STD_ZVAL(tmp);
-	mailparse_get_part_data(part, tmp TSRMLS_CC);
-	add_property_zval(object, "data", tmp);
-	Z_DELREF_P(tmp);
+	mailparse_get_part_data(part, &tmp TSRMLS_CC);
+	add_property_zval(object, "data", &tmp);
+	Z_DELREF_P(&tmp);
 
 	return SUCCESS;
 }
 
 static int mailparse_mimemessage_export(php_mimepart *part, zval *object TSRMLS_DC)
 {
-	zval *zpart;
+	zval zpart;
 
-	zend_list_addref(part->rsrc_id);
+	part->rsrc->gc.refcount++;
 
-	MAKE_STD_ZVAL(zpart);
-	php_mimepart_to_zval(zpart, part);
+	php_mimepart_to_zval(&zpart, part->rsrc);
 
 	object_init_ex(object, mimemsg_class_entry);
 
-	Z_SET_ISREF_TO_P(object, 1);
+	ZVAL_MAKE_REF(object);
 	Z_SET_REFCOUNT_P(object, 1);
```

Comment:
I believe these codes is out of data, make an object a reference make no sense..